tests/unittests/test_handler/test_handler_ntp.py

```diff
@@ -452,24 +500,48 @@ def test_ntp_the_whole_package(self, m_sysd, m_select, m_subp):
             confpath = ntpconfig['confpath']
             service_name = ntpconfig['service_name']
             m_select.return_value = ntpconfig
-            pools = cc_ntp.generate_server_names(mycloud.distro.name)
-            # force uses systemd path
-            m_sysd.return_value = True
-            with mock.patch('cloudinit.config.cc_ntp.util') as m_util:
-                # allow use of util.mergemanydict
-                m_util.mergemanydict.side_effect = util.mergemanydict
-                # default client is present
-                m_subp.which.return_value = True
-                # use the config 'enabled' value
-                m_util.is_false.return_value = util.is_false(
-                    cfg['ntp']['enabled'])
-                cc_ntp.handle('notimportant', cfg, mycloud, None, None)
-                m_subp.subp.assert_called_with(
-                    ['systemctl', 'reload-or-restart',
-                     service_name], capture=True)
-            self.assertEqual(
-                "servers []\npools {0}\n".format(pools),
-                util.load_file(confpath))
```

Comment:
We can drop the big else/if and code duplication with

```
hosts = cc_ntp.generate_server_names(mycloud.distro.name)
use_systemd = True
expected_service_call = ['systemctl', 'reload-or-restart', service_name]
expected_content = "servers []\npools {0}\n".format(hosts)
if distro == 'alpine':
     uses_systemd = False
     expected_service_call = ['service' , service_name, 'restart']
     expected_cpmtemt = "servers {0}\npools []\n".format(hosts)

m_sysd.return_value = uses_systemd
...
m_subp.subp.assert_called_with(expected_service_call, capture=True)
self.assertEqual(expected_content, util.load_file(confpath))

```

Reply:
Done